e2e/e2e.py

```diff
@@ -0,0 +1,306 @@
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.firefox.options import Options
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+import random
+import string
+import time
+
+USER_RESPONSE_TIME=0.2 # A typical user cannot click faster than once every 200ms
+
+def fill_field(driver, selector, content):
+    try:
+        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
+        input = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector)))
+        input.clear()
+        input.send_keys(content)
+    except Exception as e:
+        print("Failed when filling in field '{}'".format(selector))
+        raise e
+
+def click_button(driver, selector):
+    try:
+        print("Clicking button: {}".format(selector))
+        # Force a small wait to mimick a real user. This is to stabilize these tests a bit more
+        time.sleep(USER_RESPONSE_TIME)
+        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
+        button = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector)))
+        button.click()
+    except Exception as e:
+        print("Failed when clicking on button '{}'".format(selector))
+        raise e
+
+def click_button_after_hover(driver, hover_selector, button_selector):
+    try:
+        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
+        hover = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, hover_selector)))
+        actions = ActionChains(driver)
+        actions.move_to_element(hover).perform()
+        driver.find_element(By.CSS_SELECTOR, button_selector).click()
```

Comment:
j'ai eu à faire ça mais je suis pas sûr des cas concrêt dans lesquels c'est nécessaire, quels sont-ils (en théorie et concrêtement) ?

Reply:
typiquement c'est pour les menu du header: tu peux pas cliquer sur un élément qui n'est pas visible. Il faut donc aller hover sur "Play offline" si tu veux cliquer sur "Learn the rules"

Reply:
Faut pas simplement clicker sur le menu déroulant, puis son élément ?